pkg/manifests/nmstateconfig.go

```diff
@@ -0,0 +1,71 @@
+package manifests
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+)
+
+func getNMStateConfig() aiv1beta1.NMStateConfig {
+	nmStateConfigData, err := os.ReadFile("./manifests/nmstateconfig.yaml")
+	if err != nil {
+		fmt.Errorf("Error reading pull secret: %w", err)
+	}
+	var nmStateConfig aiv1beta1.NMStateConfig
+	if err := yaml.Unmarshal(nmStateConfigData, &nmStateConfig); err != nil {
+		fmt.Errorf("Error unmarshalling pull secret: %w", err)
+	}
+	return nmStateConfig
+}
+
+func validateNMStateConfigAndInfraEnv(nmStateConfig aiv1beta1.NMStateConfig, infraEnv aiv1beta1.InfraEnv) (bool, error) {
+	if len(nmStateConfig.ObjectMeta.Labels) == 0 {
+		return false, errors.Errorf("NMStateConfig does not have any labels set")
+	}
+
+	if len(infraEnv.Spec.NMStateConfigLabelSelector.MatchLabels) == 0 {
+		return false, errors.Errorf("Infra env does not have any labels set with NMStateConfigLabelSelector.MatchLabels")
+	}
+
+	if !reflect.DeepEqual(infraEnv.Spec.NMStateConfigLabelSelector.MatchLabels, nmStateConfig.ObjectMeta.Labels) {
+		return false, errors.Errorf("Infra env and NMStateConfig labels do not match")
+	}
+
+	return true, nil
+}
+
+func buildMacInterfaceMap(nmStateConfig aiv1beta1.NMStateConfig) models.MacInterfaceMap {
+	macInterfaceMap := make(models.MacInterfaceMap, 0, len(nmStateConfig.Spec.Interfaces))
+	for _, cfg := range nmStateConfig.Spec.Interfaces {
+		// ToDo: Use logging
+		// fmt.Println("adding MAC interface map to host static network config - Name: %s, MacAddress: %s ,",
+		// 	cfg.Name, cfg.MacAddress)
+		macInterfaceMap = append(macInterfaceMap, &models.MacInterfaceMapItems0{
+			MacAddress:     cfg.MacAddress,
+			LogicalNicName: cfg.Name,
+		})
+	}
+	return macInterfaceMap
+}
+
+func ProcessNMStateConfig(infraEnv aiv1beta1.InfraEnv) ([]*models.HostStaticNetworkConfig, error) {
+
+	nmStateConfig := getNMStateConfig()
+
+	ok, err := validateNMStateConfigAndInfraEnv(nmStateConfig, infraEnv)
+	if !ok {
+		return nil, err
+	}
+
+	var staticNetworkConfig []*models.HostStaticNetworkConfig
```

Comment:
I may be missing something (it may help to see an example nmstateconfig.yaml manifest) but as  ProcessNMStateConfig() is only called once, will there always be only one manifest file that would encompass all hosts?

Reply:
Here is an example of nmstateconfig https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/crds/nmstate.yaml

Reply:
Thanks, yeah I saw that one but I think that example is just for one host?

Reply:
Why a minimum? Is it mandatory?

Reply:
We are using the nmstateconfig to parse the IP and use that IP to be used for node0. Hence, at least we expect a minimum one nmstateconfig

Reply:
What I see from the mentioned example it's that there is one node configured with a static IP. Is that maybe the case we're tring to cover (ie https://github.com/openshift/openshift-docs/blob/main/modules/ztp-configuring-a-static-ip.adoc) ? I'm trying to figure it out what are the expectations and constraint, given that nmstate config could let you tweak many aspects.